### DISCUSSION TITLE
How to add the year automatically in the copyright part

### DISCUSSION THREAD

**Discussion by IvyMasterDiego** [squidfunk/mkdocs-material#4969] [Q&A] [answered]
I was thinking that every time a year passes it would be nice if this was automatic, but I don't know if there is a way to do it or if it isn't. If you could tell me please👉👈

**Comment by squidfunk** [comment]
You might be able to use the [macros plugin](https://github.com/fralau/mkdocs_macros_plugin), but I'm not sure whether it also applies to values in `mkdocs.yml`.

**Comment by jeremy-feng** [comment]
Add a simple javascript into your copyright config:

`<p> Copyright &copy; 2022-<script>document.write(new Date().getFullYear())</script> Your Name All Rights Reserved</p>
`

![image](https://user-images.githubusercontent.com/44312563/215980960-95d3acd4-6b94-4633-9887-09e0594e289a.png)

**Reply by shenweiyan** [reply]
This approach can cause some issues when the page is refreshed. It will display the year on the page, and we need to refresh the page again to restore the normal content.

**Reply by squidfunk** [reply, marked as answer]
Simplest solution: ask ChatGPT to write a [hook](https://www.mkdocs.org/user-guide/configuration/#hooks) for you:

``` py
from datetime import datetime
def on_config(config, **kwargs):
    config.copyright = f"Copyright © {datetime.now().year} Your Name"
```

**Reply by hellt** [reply]
nice one with the hook, because I think the macros plugin does not template the mkdocs.yml or partials

**Reply by kamilkrzyskow** [reply]
You could also keep the copyright string in `mkdocs.yml` and only specify the location where to put the year, then the hook would format the string from `mkdocs.yml`. That's how I added it in the [`mkdocs-nype` theme](https://github.com/nypesap/mkdocs-nype) to disconnect the "dynamic year logic" from the actual content.

```py
year: str = str(datetime.now().year)
config.copyright = config.copyright.format(year=year)
```

```yaml
copyright: Copyright &copy; {year} Nype
```

**Reply by hellt** [reply]
this is even better, thanks @kamilkrzyskow
